core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -84,6 +89,25 @@ class BaseSnapshot implements Snapshot {
     this.manifestListLocation = manifestList;
   }
 
+  BaseSnapshot(long sequenceNumber,
+               long snapshotId,
+               Long parentId,
+               long timestampMillis,
+               String operation,
+               Map<String, String> summary,
+               Integer schemaId,
+               String manifestList) {
+    this.io = null;
+    this.sequenceNumber = sequenceNumber;
+    this.snapshotId = snapshotId;
+    this.parentId = parentId;
+    this.timestampMillis = timestampMillis;
+    this.operation = operation;
+    this.summary = summary;
+    this.schemaId = schemaId;
+    this.manifestListLocation = manifestList;
+  }
```

Comment:
This constructor is not presently used. In most situations, it's ok to simply pass `null` as the FileIO.

I didn't update the specific constructor used in an attempt to make this easier to cherry-pick.